File: search-insert-position/search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int leftSide = 0; 
        int rightSide = nums.size(); 
        
        while (leftSide < rightSide) {
            int mid = (leftSide + rightSide) / 2; 
            if (nums[mid] == target)
                return mid; 
            if (nums[mid] < target)
                leftSide = mid + 1; 
            if (nums[mid] > target)
                rightSide = mid; 
        }
        return leftSide; 
    }
};

File: search-insert-position/search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int index = 0;
        for(int i = 0; i< nums.length; i++){
            if(nums[i]==target){
                return i;
            }
            if(nums[i] < target){
                index = i+1;
            }
            else break;
        }
        return index;
    }
}